_posts/en/newsletters/2023-03-08-newsletter.md

```diff
@@ -0,0 +1,226 @@
+        week).
+
+      - The *freeze leafscript* contains any authorization conditions
+        Alice wants to specify (including none at all) and
+        an `OP_FORWARD_DESTINATION` opcode.  At the time she creates
+        this leafscript, she also chooses her more-trusted authorization
+        conditions (such as requiring multiple signatures from multiple
+        cold wallets and hardware signing devices).  She provides the
+        opcode a commitment to those conditions in the form of a hash
+        digest.
+
+    - *Alice triggers an unvaulting* by spending the output received to
+      the above script tree (using it as an input) and choosing the
+      trigger leafscript.  At this time, she provides two additional
+      parameters to the `OP_TRIGGER_FORWARD` opcode, the index of the
+      output which will receive this input's funds and a hash-based
+      commitment to how she wants to be able to spend the funds later.
+      The opcode verifies that the indicated output of this transaction
+      pays a P2TR output with a script tree similar to the one being spent except that the
+      trigger leafscript is replaced with a script using an
+      `OP_CHECKSEQUENCEVERIFY` (CSV) relative delay equal to the delay
+      specified previously (e.g., 1000 blocks) and an
+      `OP_FORWARD_OUTPUTS` opcode which includes Alice's commitment
+      hash.  The method of reconstructing the script tree is similar to
+      an earlier [covenant][topic covenants] proposal,
+      `OP_TAPLEAF_UPDATE_VERIFY` (see [Newsletter #166][news166 tluv]).
+
+    - *Alice completes the unvaulting* by waiting until the relative
+      timelock has expired and then spending the unvaulting output,
+      choosing the tapleaf with the `OP_FORWARD_OUTPUTS` opcode.  The
+      opcode verifies that the spending transaction's output amounts and
+      script’s hash to the same commitment Alice made in the previous
+      transaction.  In this case, Alice has successfully deposited funds
+      to a vault, begun an unvaulting, been forced to wait at least
+      1,000 blocks to allow her monitoring programs to verify she really
+      did want to spend the funds to the specified outputs, and
+      completed the spend.
+
+    - If something goes wrong, *Alice freezes the funds*.  She can do
+      this at any time from the moment she deposits funds in the vault
+      up until an unvaulting is completed.  To freeze funds, she simply
+      chooses to spend the freeze leafscript from the output of
+      either the vaulting or trigger transactions.  Recall
+      that Alice explicitly placed the freeze leafscript in the vaulting
+      transaction, and note that it was implicitly carried over by the
+      trigger transaction which initiated the unvaulting.
+
+  One of the advantages to users of this approach over the original
+  `OP_VAULT` design is that the freeze leafscript can contain any
+  authorization conditions Alice wants to specify.  In the `OP_VAULT`
```

Comment:
For what it's worth, this was the case with BIP 345 before Sanders made these suggestions. Search for [`optional auth. scriptPubKey](https://github.com/jamesob/bips/blob/jamesob-23-02-opvault/bip-vaults.mediawiki#user-content-Parameters) in the current text of the BIP. Greg's rework doesn't add this in.

Reply:
Was that the result of some previous change in the design?  In our original coverage of the idea, we linked to a post by AJ where he mentioned that as a concern, see https://bitcoinops.org/en/newsletters/2023/01/18/#proposal-for-new-vault-specific-opcodes where we wrote,

> Towns [notes](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2023-January/021328.html) that [...] any third party who learns the address can also freeze the user’s funds (although they’ll have to pay a transaction fee to do so), creating an inconvenience for the user.

If it was a previous change to the proposal that added the ability to allow requiring arbitrary authorization for freezing, then I think we're technically correct to say the an advantage of Greg's approach over the original approach (as posted to the ML on Jan 9th) is this ability.

However, I also get that it's misleading in the sense that it's not an advantage over what you wanted to ship at the time Greg suggested it.  Would you like us to print a correction in next week's newsletter?